src/lib/phdi-building-blocks/phdi_building_blocks/standardize.py

```diff
@@ -167,35 +246,104 @@ def standardize_country(
     return standard
 
 
-def standardize_patient_phone(
-    bundle: dict, standardize: Callable = standardize_phone
-) -> dict:
-    """Given a FHIR bundle and a standardization function, standardize the phone numbers
-    in all patient resources in the bundle. By default the standardize_phone function
-    defined in this module is used, but the user is free to provide their own function
-    specifying a standardization process as long as it accepts and returns a string."""
+def phone_truncation_standardization(raw_phone: str, *args) -> str:
+    """
+    Truncate a given phone number string to exactly ten digits, after
+    removing any special characters. For example,
+    standardize_phone("(555) 555-1212") would return '5555551212'. If
+    the provided phone number is less than ten digits, then the
+    function simply returns None, as this is not a valid phone number.
+    :param str raw_phone: The phone number to standardize
+    :return: The truncated phone number
+    :rtype: str
+    """
+    raw_phone = [x for x in raw_phone if x.isnumeric()]
+    raw_phone = "".join(raw_phone)
+    if len(raw_phone) < 10:
+        raw_phone = None
+    elif len(raw_phone) > 10:
+        raw_phone = raw_phone[:10]
+    return raw_phone
+
 
+def standardize_all_phones_in_bundle(
+    bundle: dict,
+    standardization_mode: str = "truncation",
+    add_phone_metrics: bool = True,
+    overwrite: bool = True,
+) -> dict:
+    """
+    Given a FHIR bundle and a type of phone number standardization,
+    transform all phone numberes for all patient resources in the
+    bundle. The default mode of standardization is "truncation". Also,
+    optionally store tracking data around whether any phone numbers
+    were transformed in an extension array directly on the patient.
+    :param dict bundle: The FHIR bundle to standardize patients in
+    :param str standardization_mode: The type of standardization to
+    perform. Valid options include: "truncation"
+    :param bool add_phone_metrics: Whether to store tracking metrics
+    for changed phone numbers. Default is yes.
+    :param bool overwrite: Whether to overwrite the original data
+    with the transformed values. Default is yes.
+    :return: The bundle with all phone numbers transformed
+    :rtype: dict
+    """
+    if not overwrite:
+        bundle = copy.deepcopy(bundle)
     for resource in find_patient_resources(bundle):
         patient = resource.get("resource")
+        if standardization_mode == "truncation":
+            standardize_phone_numbers_for_patient(
+                patient, phone_truncation_standardization, add_phone_metrics
+            )
+        elif standardization_mode == "country":
+            standardize_phone_numbers_for_patient(
+                patient, phone_country_standardization, add_phone_metrics
+            )
```

Comment:
I think this is another good reason for not going with this approach. Any time we add a method of standardization, we have to update it in a _bunch_ of places, which makes the code harder to read. And here, we could end up with a long list of `if/elif` statements. If we end up using this approach of specifying a helper function to perform the standardization, then we absolutely need to make use of the `FUNC_MAP` variable.